drivers/temperature/adt7420/adt7420.h

```diff
@@ -42,92 +42,151 @@
-
-/* ADT7420 registers */
-#define ADT7420_REG_TEMP_MSB		0x00 // Temperature value MSB
-#define ADT7420_REG_TEMP_LSB		0x01 // Temperature value LSB
-#define ADT7420_REG_STATUS		0x02 // Status
-#define ADT7420_REG_CONFIG		0x03 // Configuration
-#define ADT7420_REG_T_HIGH_MSB		0x04 // Temperature HIGH setpoint MSB
-#define ADT7420_REG_T_HIGH_LSB		0x05 // Temperature HIGH setpoint LSB
-#define ADT7420_REG_T_LOW_MSB		0x06 // Temperature LOW setpoint MSB
-#define ADT7420_REG_T_LOW_LSB		0x07 // Temperature LOW setpoint LSB
-#define ADT7420_REG_T_CRIT_MSB		0x08 // Temperature CRIT setpoint MSB
-#define ADT7420_REG_T_CRIT_LSB		0x09 // Temperature CRIT setpoint LSB
-#define ADT7420_REG_HIST		0x0A // Temperature HYST setpoint
-#define ADT7420_REG_ID			0x0B // ID
-#define ADT7420_REG_RESET		0x2F // Software reset
+#define ADT7420_A0_PIN(x)	            (((x) & 0x1) << 0) // I2C Serial Bus Address Selection Pin
+#define ADT7420_A1_PIN(x)	            (((x) & 0x1) << 1) // I2C Serial Bus Address Selection Pin
+#define ADT7420_ADDRESS(x,y)	        (0x48 + ADT7420_A1_PIN(x) + ADT7420_A0_PIN(y))
+
+/* ADT7320  (SPI) registers */
+#define ADT7320_REG_STATUS			    0x00 // Status
+#define ADT7320_REG_CONFIG			    0x01 // Configuration
+#define ADT7320_REG_TEMP    		    0x02 // Temperature value
+#define ADT7320_REG_ID				    0x03 // ID
+#define ADT7320_REG_T_CRIT			    0x04 // Temperature CRIT setpoint (147'C)
+#define ADT7320_REG_HIST			    0x05 // Temperature HYST setpoint (5'C)
+#define ADT7320_REG_T_HIGH    		    0x06 // Temperature HIGH setpoint (64'C)
+#define ADT7320_REG_T_LOW    		    0x07 // Temperature LOW setpoint (10'C)
+
+/* ADT7320 SPI command byte */
+#define ADT7320_WRITE_MASK_CMD          0b00111000 // SPI write command
+#define ADT7320_READ_CMD                0b01000000 // SPI read command
+
+/* ADT7420 (I2C) registers */
+#define ADT7420_REG_TEMP_MSB		    0x00 // Temperature value MSB
+#define ADT7420_REG_TEMP_LSB		    0x01 // Temperature value LSB
+#define ADT7420_REG_STATUS			    0x02 // Status
+#define ADT7420_REG_CONFIG			    0x03 // Configuration
+#define ADT7420_REG_T_HIGH_MSB		    0x04 // Temperature HIGH setpoint MSB
+#define ADT7420_REG_T_HIGH_LSB		    0x05 // Temperature HIGH setpoint LSB
+#define ADT7420_REG_T_LOW_MSB		    0x06 // Temperature LOW setpoint MSB
+#define ADT7420_REG_T_LOW_LSB		    0x07 // Temperature LOW setpoint LSB
+#define ADT7420_REG_T_CRIT_MSB		    0x08 // Temperature CRIT setpoint MSB
+#define ADT7420_REG_T_CRIT_LSB		    0x09 // Temperature CRIT setpoint LSB
+#define ADT7420_REG_HIST			    0x0A // Temperature HYST setpoint
+#define ADT7420_REG_ID				    0x0B // ID
+#define ADT7420_REG_RESET			    0x2F // Software reset
 
 /* ADT7420_REG_STATUS definition */
-#define ADT7420_STATUS_T_LOW		(1 << 4)
-#define ADT7420_STATUS_T_HIGH		(1 << 5)
-#define ADT7420_STATUS_T_CRIT		(1 << 6)
-#define ADT7420_STATUS_RDY		(1 << 7)
+#define ADT7420_STATUS_T_LOW		    (1 << 4)
```

Comment:
Use BIT(4)
Same comment for all the ones below

Reply:
Implemented the functionality using BIT() function from util.h